src/ol_concourse/pipelines/infrastructure/heroku_dyno_reset/pipeline.py

```diff
@@ -0,0 +1,347 @@
+                        """
+                        ),
+                    ],
+                ),
+            ),
+        )
+    )
+
+    qa_jobs.append(
+        Job(
+            name=f"reset-dynos-nonprod-{app_name}",
+            build_log_retention={"days": 30},
+            plan=task_list,
+        )
+    )
+
+qa_combined_fragment = PipelineFragment(
+    resources=[qa_build_schedule],
+    jobs=qa_jobs,
+)
+
+production_jobs = []
+for app_name, app_config in production_dyno_map.items():
+    heroku_api_key = read_yaml_secrets(
+        Path().joinpath("heroku", f"secrets.{app_config['owner']}.yaml")
+    )["apiKey"]
+
+    task_list = [GetStep(get=production_build_schedule.name, trigger=True)]
+
+    for dyno_config in app_config["dyno_list"]:
+        task_list.append(
+            TaskStep(
+                task=Identifier(f"reset-{dyno_config.split('=')[0]}"),
+                config=TaskConfig(
+                    platform=Platform.linux,
+                    image_resource=heroku_cli_resource,
+                    inputs=[],
+                    outputs=[],
+                    run=Command(
+                        path="sh",
+                        args=[
+                            "-ec",
+                            textwrap.dedent(
+                                f"""
+                            export HEROKU_API_KEY={heroku_api_key}
+                            heroku dyno:scale -a {app_name} {dyno_config}
+                            """
+                            ),
+                        ],
+                    ),
+                ),
+            )
+        )
+    task_list.append(
+        TaskStep(
+            task=Identifier("show-dyno-config"),
+            config=TaskConfig(
+                platform=Platform.linux,
+                image_resource=heroku_cli_resource,
+                inputs=[],
+                outputs=[],
+                run=Command(
+                    path="sh",
+                    args=[
+                        "-ec",
+                        textwrap.dedent(
+                            f"""
+                        export HEROKU_API_KEY={heroku_api_key}
+                        heroku dyno:type -a {app_name}
+                        """
+                        ),
+                    ],
+                ),
+            ),
+        )
+    )
+
+    production_jobs.append(
+        Job(
+            name=f"reset-dynos-prod-{app_name}",
+            build_log_retention={"days": 30},
+            plan=task_list,
+        )
+    )
+
+production_combined_fragment = PipelineFragment(
+    resources=[production_build_schedule],
+    jobs=production_jobs,
+)
+
+fully_combined_fragment = PipelineFragment(
+    resource_types=qa_combined_fragment.resource_types
+    + production_combined_fragment.resource_types,
+    resources=qa_combined_fragment.resources + production_combined_fragment.resources,
+    jobs=qa_combined_fragment.jobs + production_combined_fragment.jobs,
+)
```

Comment:
For this you can do `PipelineFragment.combine_fragments(qa_combined_fragment, production_combined_fragment)`